### ISSUE TITLE
Better custom middleware?

### ISSUE THREAD

**Issue by slavafomin** [balderdashy/sails#2603]
Hello!

According [to the docs] (http://sailsjs.org/#/documentation/concepts/Middleware), custom middleware should be defined inside of a «**config/http.js**» file. There is no built-in or otherwise conventional way to define middleware in a separate files (correct me if I'm wrong please). I think this is a major drawback, cause business-logic should not be contained in the *config* directory.

The better approach will be to store custom middleware inside of the project in the «middleware» directory. Such middleware files can be autoloaded exactly like controllers are. The name of the middleware will be defined by the name of the file without extension.

Does it makes sense? Thank you!

**Comment by randallmeeker** [comment]
I agree that if we were to follow convention then there should be middlerware folder in the api just like controllers, responses, models, services ect . . . .

**Comment by thomasfr** [comment]
yep +1

This is somewhat cumbersome at the moment

**Comment by mikermcneil** [comment]
Thanks for posting, @slavafomin.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by mikermcneil** [comment]
> There is no built-in or otherwise conventional way to define middleware in a separate files (correct me if I'm wrong please).

(btw @slavafomin I'd recommend using `require()` for this)

**Comment by slavafomin** [comment]
Hello!

@mikermcneil could you elaborate please on your suggestion? I'm not sure I get it.

And why are you closing this? I think this feature request is still valid. Thank you.
